sentinel-core/src/main/java/com/alibaba/csp/sentinel/slots/block/authority/AuthorityRuleManager.java

```diff
@@ -93,25 +93,21 @@ public static boolean hasConfig(String resource) {
 
         @Override
         public void configUpdate(List<AuthorityRule> conf) {
-            Map<String, Set<AuthorityRule>> rules = loadAuthorityConf(conf);
-
-            authorityRules.clear();
-            if (rules != null) {
-                authorityRules.putAll(rules);
-            }
+            authorityRules = loadAuthorityConf(conf);
```

Comment:
if you want to assign the new value to `authorityRules`, I think you need to declare it as `volatile`, so that the other threads can see the changes immediately.